worker/uploader.go

```diff
@@ -232,17 +248,28 @@ func (u *uploader) estimate() float64 {
 
 // execute executes the sector upload request, if the upload was successful it
 // returns the time it took to upload the sector to the host
-func (u *uploader) execute(req *sectorUploadReq) (time.Duration, error) {
+func (u *uploader) execute(req *sectorUploadReq) (_ time.Duration, err error) {
 	// grab fields
 	u.mu.Lock()
 	host := u.host
 	fcid := u.fcid
 	u.mu.Unlock()
 
+	// wrap cause
+	defer func() {
```

Comment:
This seems redundant considering that this method doesn't check `req.sector.ctx` and if any children do it they should probably return that error. I'd like to avoid ending up with this construct all over the place.

Reply:
So it turns out that it's not redundant, I added it because I noticed in the logs some cases weren't treated properly. I don't like it more than you but in this case I feel it's a good sanity check because we use the context cause together with `errSectorUploadFinished` to decide what the penalty should be. If we ever miss the cause we mistreat the case entirely. The case where it is not added is where we are in the middle of a call to the bus, the bus client will return context canceled and doesn't return the cause. I considered tweaking the client (or whatever the alternative solution would be) but figured this wrap was better.